### PR TITLE
SME review

### PR DESCRIPTION
**Pull request by shin19991207**
According to https://github.com/OpenLiberty/draft-guide-grpc-intro/issues/10:

### https://draft-openlibertyio.mybluemix.net/guides/grpc-intro.html#what-is-grpc

- [x] In "For each service defined in a .proto file, gRPC uses it to generate the skeleton code for users to implement and extend. "
"it" would be better as "the definition"

- [x] "much faster" -> "faster"

- [x] "either supported languages" -> other supported languages

### https://draft-openlibertyio.mybluemix.net/guides/grpc-intro.html#what-youll-learn

- [x] "client and server services" -> "services and their clients"    i.e. is client code a service? is 'server service' odd sounding? 

### https://draft-openlibertyio.mybluemix.net/guides/grpc-intro.html#try-what-youll-build

- [x] Opened https://github.com/OpenLiberty/ci.maven/issues/1584 to address the below issue

> When issue `mvn -pl system liberty:run`, the message shows the copying of server.xml twice:
> 
> ```
> **[INFO] Copying 1 file to /Users/hutchig/git/draft-guide-grpc-intro/finish/system/target/liberty/wlp/usr/servers/defaultServer
> [INFO] CWWKM2144I: Update server configuration file server.xml from /Users/hutchig/git/draft-guide-grpc-intro/finish**/system/src/main/liberty/config/server.xml.
> [INFO] CWWKM2185I: The liberty-maven-plugin configuration parameter "appsDirectory" value defaults to "apps".
> [INFO] CWWKM2160I: Installing application guide-grpc-intro-system.war.xml.
> **[INFO] Copying 1 file to /Users/hutchig/git/draft-guide-grpc-intro/finish/system/target/liberty/wlp/usr/servers/defaultServer
> [INFO] CWWKM2144I: Update server configuration file server.xml from /Users/hutchig/git/draft-guide-grpc-intro/finish**/system/src/main/liberty/config/server.xml.
> ```

- [x] https://github.com/OpenLiberty/draft-guide-grpc-intro/issues/10#issuecomment-1238441784
> `[INFO] [WARNING ] SRVE0190E: File not found: /favicon.ico` appears every time http://localhost:9080/ is hit, it also appears each time http://localhost:9081/query/properties/os.name is hit on a new browser. 

### https://draft-openlibertyio.mybluemix.net/guides/grpc-intro.html#implementing-server-streaming-call

- [x] Update `getPropertyName` to `getPropertyPrefix`

- [x] "Then, call the getPropertiesServer()" -> "Then, calls the getPropertiesServer()"

- [x] Same problem above with "shutdown the channel and returns"  ....mixed form of expression instructions or descriptions of what needs doing or is being done.

### https://draft-openlibertyio.mybluemix.net/guides/grpc-intro.html#implementing-client-streaming-call

- [x] "to implement of" -> "to implement"  

### https://draft-openlibertyio.mybluemix.net/guides/grpc-intro.html#implementing-client-streaming-call

- [x] "properties started with user." -> "properties with property names prefixed by 'user.' "

### https://draft-openlibertyio.mybluemix.net/guides/grpc-intro.html#implementing-bidirectional-streaming-call

- [x] "returns an instance"  and "close the server" so the same mix of description and instruction forms of words when both bits of function have the same initial location

### https://draft-openlibertyio.mybluemix.net/guides/grpc-intro.html#running-the-tests

- [x] It would useful to identify WHICH session one can press return in (as there are multiple session in dev mode
at this point) query.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -154,14 +154,14 @@ include::finish/systemproto/pom.xml[]
...
 
-The [hotspot=getProperty file=0]`getProperty` rpc defines the unary call. In this call, the client side sends a `SystemPropertyName` message to the server side that returns back a `SystemPropertyValue` message with the property value. The [hotspot=SystemPropertyName file=0]`SystemPropertyName` and [hotspot=SystemPropertyValue file=0]`SystemPropertyValue` message types define that the `propertyName` and `propertyValue` must be string.
+The [hotspot=getProperty file=0]`getProperty` rpc defines the unary call. In this call, the client side sends a `SystemPropertyPrefix` message to the server side that returns back a `SystemPropertyValue` message with the property value. The [hotspot=SystemPropertyPrefix file=0]`SystemPropertyPrefix` and [hotspot=SystemPropertyValue file=0]`SystemPropertyValue` message types define that the `propertyPrefix` and `propertyValue` must be string.
```
Comment (hutchig) [inline README.adoc]: Why can't it send a 'name' - what if more than one property has the prefix?

File: README.adoc [hunk 2]
```diff
@@ -154,14 +154,14 @@ include::finish/systemproto/pom.xml[]
...
 
-The [hotspot=getPropertiesServer file=0]`getPropertiesServer` rpc defines the server streaming call. The client side sends a `SystemPropertyName` message to the server side. The server returns back a stream of `SystemProperty` messages. Each [hotspot=SystemProperty file=0]`SystemProperty` message contains a `propertyName` and a `propertyValue` strings.
+The [hotspot=getPropertiesServer file=0]`getPropertiesServer` rpc defines the server streaming call. The client side sends a `SystemPropertyPrefix` message to the server side. The server returns back a stream of `SystemProperty` messages. Each [hotspot=SystemProperty file=0]`SystemProperty` message contains a `propertyPrefix` and a `propertyValue` strings.
```
Comment (hutchig) [inline README.adoc]: A single value is not associated with a prefix but a single property (which has a name)

File: README.adoc [hunk 3]
```diff
@@ -230,7 +230,7 @@ include::finish/system/src/main/java/io/openliberty/guides/system/SystemService.
...
+The [hotspot=getProperty file=0]`getProperty()` method implements the rpc unary call defined in the proto file. Use the [hotspot=pPrefix file=0]`getPropertyPrefix()` getter method that is generated by gRPC to retrieve the property name from the client, and store into the `pPrefix` variable. Get and store the System property value into the [hotspot=pValue file=0]`pValue` variable. Use the the gRPC APIs to create a [hotspot=response file=0]`SystemPropertyValue` message that its type is defined in the `SystemService.proto` file. Then, send the message to the client service through the `StreamObserver` by using its [hotspot=next file=0]`onNext()` and [hotspot=complete file=0]`onComplete()` methods.
```
Comment (hutchig) [inline README.adoc]: This was correctish and is now incorrect. I thin getPropertyName would be better as getProperty here NOT anything to do with prefix. (i.e. my previous comment about Prefix did not apply to the singular property calls.)

File: README.adoc [hunk 4]
```diff
@@ -261,7 +261,7 @@ PropertiesResource.java
...
 ----
-The `PropertiesResource` class provides RESTful endpoints to interact with the `system` service. The [hotspot=unary file=2]`/query/properties/${propertyName}` endpoint uses the unary service call to get the property value from the `system` service. The endpoint creates a [hotspot=createChannel1 file=2]`channel`, uses the channel to create a client by the [hotspot=createClient1 file=2]`SystemServiceGrpc.newBlockingStub()` API, uses the client to get the property value, shutdowns the channel, and then returns the value that the `system` service responses immediately.
+The `PropertiesResource` class provides RESTful endpoints to interact with the `system` service. The [hotspot=unary file=2]`/query/properties/${propertyPrefix}` endpoint uses the unary service call to get the property value from the `system` service. The endpoint creates a [hotspot=createChannel1 file=2]`channel`, uses the channel to create a client by the [hotspot=createClient1 file=2]`SystemServiceGrpc.newBlockingStub()` API, uses the client to get the property value, shutdowns the channel, and then returns the value that the `system` service responses immediately.
```
Comment (hutchig) [inline README.adoc]: it is not getting a property name or prix but just a property:
```
'/query/properties//${property}"
```
??

File: README.adoc [hunk 5]
```diff
@@ -313,7 +313,7 @@ SystemService.java
...
+The [hotspot=getPropertiesServer file=0]`getPropertiesServer()` method implements server streaming rpc call. Use the [hotspot=prefix file=0]`getPropertyPrefix()` getter method to retrieve the property prefix from the client. Filter out the properties that starts with the [hotspot=filter file=0]`prefix`. For each property, build a [hotspot=serverMessage file=0]`SystemProperty` message and stream the message to the client through the `StreamObserver` by using its [hotspot=serverNext1 file=0]`onNext()` method. When all properties are streamed, finish the streaming by calling the [hotspot=serverComplete file=0]`onComplete()` method.
```
Comment (hutchig) [inline README.adoc:316]: getPropertiesWithPrefix
??

File: README.adoc [hunk 6]
```diff
@@ -330,7 +330,7 @@ PropertiesResource.java
...
+The endpoint creates a [hotspot=createChannel2 file=1]`channel` to the `system` service and a [hotspot=createClient2 file=1]`client` by using the `SystemServiceGrpc.newStub()` API. Then, calls the [hotspot=getPropertiesServer file=1]`getPropertiesServer()` method with an implementation of the `StreamObserver` interface. The [hotspot=onNext1 file=1]`onNext()` method receives messages streaming from the server individually and stores them into the `properties` placeholder. After all properties are received, shutdowns the [hotspot=closeConnection file=1]`channel` and returns the placeholder. Because the rpc call is asynchronous, use a [hotspot=countDownLatch1 hotspot=countDownLatch2 hotspot=countDownLatch3 file=1]`CountDownLatch` to synchronize the streaming flow.
```
Comment (hutchig) [inline README.adoc]: THe sentance "After all properties are received, shutdowns the [hotspot=closeConnection file=1]`channel` and returns the placeholder."
needs a subject like "it"  ...  it shuts down.

File: README.adoc [hunk 7]
```diff
@@ -371,7 +371,7 @@ The [hotspot=getPropertiesClient file=0]`getPropertiesClient()` method implement
 
 
 // Replace PropertiesResource
-Update the `PropertiesResource` class to implement of `/query/properties/user` endpoint of the query service.
+Update the `PropertiesResource` class to implement `/query/properties/user` endpoint of the query service.
```
Comment (hutchig) [inline README.adoc]: the

File: finish/query/src/main/java/io/openliberty/guides/query/PropertiesResource.java [hunk 8]
```diff
@@ -55,19 +55,21 @@ public class PropertiesResource {
 
     // tag::unary[]
     @GET
-    @Path("/{propertyName}")
+    @Path("/{propertyPrefix}")
```
Comment (hutchig) [inline finish/query/src/main/java/io/openliberty/guides/query/PropertiesResource.java]: Unary is Name

File: finish/query/src/main/java/io/openliberty/guides/query/PropertiesResource.java [hunk 9]
```diff
@@ -55,19 +55,21 @@ public class PropertiesResource {
...
     @Produces(MediaType.TEXT_PLAIN)
-    public String getPropertiesString(@PathParam("propertyName") String propertyName) {
+    public String getPropertiesString(
+        @PathParam("propertyPrefix") String propertyPrefix) {
```
Comment (hutchig) [inline finish/query/src/main/java/io/openliberty/guides/query/PropertiesResource.java]: Unary retains Name not prefix

File: finish/query/src/main/java/io/openliberty/guides/query/PropertiesResource.java [hunk 10]
```diff
@@ -55,19 +55,21 @@ public class PropertiesResource {
...
+        ManagedChannel channel = ManagedChannelBuilder
+                                     .forAddress(SYSTEM_HOST, SYSTEM_PORT)
                                      .usePlaintext().build();
         // end::createChannel1[]
         // tag::createClient1[]
         SystemServiceBlockingStub client = SystemServiceGrpc.newBlockingStub(channel);
         // end::createClient1[]
-        SystemPropertyName request = SystemPropertyName.newBuilder()
-                                             .setPropertyName(propertyName).build();
+        SystemPropertyPrefix request = SystemPropertyPrefix.newBuilder()
```
Comment (hutchig) [inline finish/query/src/main/java/io/openliberty/guides/query/PropertiesResource.java]: unary is name

File: finish/query/src/main/java/io/openliberty/guides/query/PropertiesResource.java [hunk 11]
```diff
@@ -55,19 +55,21 @@ public class PropertiesResource {
...
+                                             .setPropertyPrefix(propertyPrefix).build();
```
Comment (hutchig) [inline finish/query/src/main/java/io/openliberty/guides/query/PropertiesResource.java]: unary still uses name

File: finish/system/src/main/java/io/openliberty/guides/system/SystemService.java [hunk 12]
```diff
@@ -62,10 +62,10 @@ public void getProperty(
     // tag::getPropertiesServer[]
     @Override
     public void getPropertiesServer(
-        SystemPropertyName request, StreamObserver<SystemProperty> observer) {
+        SystemPropertyPrefix request, StreamObserver<SystemProperty> observer) {
```
Comment (hutchig) [inline finish/system/src/main/java/io/openliberty/guides/system/SystemService.java:66]: single value retains 'name'  (only when retrieving a set prefix is used as the filter key)

File: finish/system/src/main/java/io/openliberty/guides/system/SystemService.java [hunk 13]
```diff
@@ -139,20 +139,20 @@ public void onCompleted() {
 
     // tag::getPropertiesBidirect[]
     @Override
```
Comment (hutchig) [inline finish/system/src/main/java/io/openliberty/guides/system/SystemService.java]: The results are property names, it is only the filter/key which would be better as prefix. ie. only the thing that is used as a prefix

**Review by hutchig — CHANGES_REQUESTED** [review]

**Review by hutchig — APPROVED** [review]
Looks ok to me!

**Review by gkwan-ibm — CHANGES_REQUESTED** [review]
hi @shin19991207 

- "Creating and defining gRPC service" section
  - [x] "The `getPropertiesServer` rpc defines the server streaming call. The client side sends a `SystemPropertyName` message ..." -> "The `getPropertiesServer` rpc defines the server streaming call. The client side sends a `SystemPropertyPrefix` message ..."
  
- SystemService.proto file
  - [x] change to following
```
message SystemPropertyPrefix {
    string propertyPrefix = 1;
}
```
  - need to update the other java files that references to `getPropertyPrefix()` instead of `getPropertiesWithPrefix()`

- "Implementing server streaming call" section
  - [x] change "... Use the `getPropertiesWithPrefix()` getter method ..." to "... Use the `getPropertyPrefix()` getter method ..."

**Review by gkwan-ibm — APPROVED** [review]

**Pull request merged**